File: rockit/+rockit/GlobalOptions.m
classdef GlobalOptions < handle
  properties
    parent
  end
  methods
    function obj = GlobalOptions(varargin)
    end
  end
  methods(Static)
    function varargout = set_cmake_flags(varargin)
      global pythoncasadiinterface
      if isempty(pythoncasadiinterface)
        pythoncasadiinterface = rockit.PythonCasadiInterface;
      end
      [args,kwargs] = pythoncasadiinterface.matlab2python_arg(varargin,0,{'flags'});
      if isempty(kwargs)
        res = py.rockit.global_options.GlobalOptions.set_cmake_flags(args{:});
      else
        res = py.rockit.global_options.GlobalOptions.set_cmake_flags(args{:},pyargs(kwargs{:}));
      end
      varargout = pythoncasadiinterface.python2matlab_ret(res);
    end
    function varargout = get_cmake_flags(varargin)
      global pythoncasadiinterface
      if isempty(pythoncasadiinterface)
        pythoncasadiinterface = rockit.PythonCasadiInterface;
      end
      [args,kwargs] = pythoncasadiinterface.matlab2python_arg(varargin,0,{});
      if isempty(kwargs)
        res = py.rockit.global_options.GlobalOptions.get_cmake_flags(args{:});
      else
        res = py.rockit.global_options.GlobalOptions.get_cmake_flags(args{:},pyargs(kwargs{:}));
      end
      varargout = pythoncasadiinterface.python2matlab_ret(res);
    end
    function varargout = set_compiler(varargin)
      global pythoncasadiinterface
      if isempty(pythoncasadiinterface)
        pythoncasadiinterface = rockit.PythonCasadiInterface;
      end
      [args,kwargs] = pythoncasadiinterface.matlab2python_arg(varargin,0,{'arg'});
      if isempty(kwargs)
        res = py.rockit.global_options.GlobalOptions.set_compiler(args{:});
      else
        res = py.rockit.global_options.GlobalOptions.set_compiler(args{:},pyargs(kwargs{:}));
      end
      varargout = pythoncasadiinterface.python2matlab_ret(res);
    end
    function varargout = get_compiler(varargin)
      global pythoncasadiinterface
      if isempty(pythoncasadiinterface)
        pythoncasadiinterface = rockit.PythonCasadiInterface;
      end
      [args,kwargs] = pythoncasadiinterface.matlab2python_arg(varargin,0,{});
      if isempty(kwargs)
        res = py.rockit.global_options.GlobalOptions.get_compiler(args{:});
      else
        res = py.rockit.global_options.GlobalOptions.get_compiler(args{:},pyargs(kwargs{:}));
      end
      varargout = pythoncasadiinterface.python2matlab_ret(res);
    end
    function varargout = set_cmake_build_type(varargin)
      global pythoncasadiinterface
      if isempty(pythoncasadiinterface)
        pythoncasadiinterface = rockit.PythonCasadiInterface;
      end
      [args,kwargs] = pythoncasadiinterface.matlab2python_arg(varargin,0,{'arg'});
      if isempty(kwargs)
        res = py.rockit.global_options.GlobalOptions.set_cmake_build_type(args{:});
      else
        res = py.rockit.global_options.GlobalOptions.set_cmake_build_type(args{:},pyargs(kwargs{:}));
      end
      varargout = pythoncasadiinterface.python2matlab_ret(res);
    end
    function varargout = get_cmake_build_type(varargin)
      global pythoncasadiinterface
      if isempty(pythoncasadiinterface)
        pythoncasadiinterface = rockit.PythonCasadiInterface;
      end
      [args,kwargs] = pythoncasadiinterface.matlab2python_arg(varargin,0,{});
      if isempty(kwargs)
        res = py.rockit.global_options.GlobalOptions.get_cmake_build_type(args{:});
      else
        res = py.rockit.global_options.GlobalOptions.get_cmake_build_type(args{:},pyargs(kwargs{:}));
      end
      varargout = pythoncasadiinterface.python2matlab_ret(res);
    end
  end
end

File: rockit/global_options.py
#
#     This file is part of rockit.
#
#     rockit -- Rapid Optimal Control Kit
#     Copyright (C) 2019 MECO, KU Leuven. All rights reserved.
#
#     Rockit is free software; you can redistribute it and/or
#     modify it under the terms of the GNU Lesser General Public
#     License as published by the Free Software Foundation; either
#     version 3 of the License, or (at your option) any later version.
#
#     Rockit is distributed in the hope that it will be useful,
#     but WITHOUT ANY WARRANTY; without even the implied warranty of
#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
#     Lesser General Public License for more details.
#
#     You should have received a copy of the GNU Lesser General Public
#     License along with CasADi; if not, write to the Free Software
#     Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
#
#
cmake_flags = []
compiler = "gcc"
cmake_build_type = "Release"

class GlobalOptions:
    @staticmethod
    def set_cmake_flags(flags):
        global cmake_flags
        cmake_flags = flags
    @staticmethod
    def get_cmake_flags():
        return cmake_flags
    @staticmethod
    def set_compiler(arg):
        global compiler
        compiler = arg
    @staticmethod
    def get_compiler():
        return compiler
    @staticmethod
    def set_cmake_build_type(arg):
        global cmake_build_type
        cmake_build_type = arg
    @staticmethod
    def get_cmake_build_type():
        return cmake_build_type